app/src/lib/stores/pull-request-store.ts

```diff
@@ -251,146 +308,105 @@ export class PullRequestStore extends TypedBaseStore<GitHubRepository> {
     repository: GitHubRepository
   ): Promise<void> {
-    const repoId = repository.dbID
+    const repoDbId = repository.dbID
 
-    if (!repoId) {
-      fatalError(
+    if (repoDbId == null) {
+      return fatalError(
         "Cannot store pull requests for a repository that hasn't been inserted into the database!"
       )
-
-      return
     }
 
-    const table = this.pullRequestDatabase.pullRequests
+    const table = this._pullRequestDatabase.pullRequests
+    const prsToInsert = new Array<IPullRequest>()
+    let githubRepo: GitHubRepository | null = null
 
-    const insertablePRs = new Array<IPullRequest>()
-    for (const pr of pullRequests) {
-      let headRepo: GitHubRepository | null = null
-      if (pr.head.repo) {
-        headRepo = await this.repositoriesStore.findOrPutGitHubRepository(
+    for (const pr of apiuPullRequestsFrom) {
+      // Once the repo is found on first try, no need to keep looking
+      if (githubRepo == null && pr.head.repo != null) {
+        githubRepo = await this._repositoryStore.upsertGitHubRepository(
           repository.endpoint,
           pr.head.repo
         )
       }
 
+      if (githubRepo == null) {
+        return fatalError(
+          "The PR doesn't seem to be associated with a GitHub repository"
+        )
+      }
+
+      const githubRepoDbId = forceUnwrap(
+        'PR cannot have non-existent repo',
+        githubRepo.dbID
+      )
+
       // We know the base repo isn't null since that's where we got the PR from
       // in the first place.
-      const baseRepo = await this.repositoriesStore.findOrPutGitHubRepository(
+      const parentRepo = forceUnwrap(
+        'PR cannot have a null base repo',
+        pr.base.repo
+      )
+      const parentGitHubRepo = await this._repositoryStore.upsertGitHubRepository(
         repository.endpoint,
-        forceUnwrap('PR cannot have a null base repo', pr.base.repo)
+        parentRepo
+      )
+      const parentGitHubRepoDbId = forceUnwrap(
+        'PR cannot have a null parent database id',
+        parentGitHubRepo.dbID
       )
 
-      insertablePRs.push({
+      prsToInsert.push({
         number: pr.number,
         title: pr.title,
         createdAt: pr.created_at,
         head: {
           ref: pr.head.ref,
           sha: pr.head.sha,
-          repoId: headRepo ? headRepo.dbID! : null,
+          repoId: githubRepoDbId,
         },
         base: {
           ref: pr.base.ref,
           sha: pr.base.sha,
-          repoId: forceUnwrap('PR cannot have a null base repo', baseRepo.dbID),
+          repoId: parentGitHubRepoDbId,
         },
         author: pr.user.login,
       })
     }
 
-    await this.pullRequestDatabase.transaction('rw', table, async () => {
-      await table.clear()
-      return await table.bulkAdd(insertablePRs)
+    return this._pullRequestDatabase.transaction('rw', table, async () => {
+      const repoDbId = prsToInsert[0].base.repoId!
```

Comment:
I gather we're safe to do this because all the PRs are associated with the same repository, but can we make this a bit clearer with a comment or extracting this outside the callback?